src/containers/ServiceAccountsDropdown/ServiceAccountsDropdown.test.js

```diff
@@ -0,0 +1,277 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { fireEvent, render } from 'react-testing-library';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import ServiceAccountsDropdown from './ServiceAccountsDropdown';
+import * as API from '../../api';
+
+const props = {
+  id: 'service-accounts-dropdown'
+};
+
+const serviceAccountsByNamespace = {
+  default: {
+    default: 'id-default',
+    'service-account-1': 'id-service-account-1',
+    'service-account-2': 'id-service-account-2'
+  },
+  green: {
+    'service-account-3': 'id-service-account-3'
+  }
+};
+
+const serviceAccountsById = {
+  'id-default': {
+    metadata: {
+      name: 'default',
+      namespace: 'default',
+      uid: 'id-default'
+    }
+  },
+  'id-service-account-1': {
+    metadata: {
+      name: 'service-account-1',
+      namespace: 'default',
+      uid: 'id-service-account-1'
+    }
+  },
+  'id-service-account-2': {
+    metadata: {
+      name: 'service-account-2',
+      namespace: 'default',
+      uid: 'id-service-account-2'
+    }
+  },
+  'id-service-account-3': {
+    metadata: {
+      name: 'service-account-3',
+      namespace: 'green',
+      uid: 'id-service-account-3'
+    }
+  }
+};
+
+const namespacesByName = {
+  default: '',
+  green: '',
+  blue: ''
+};
+
+const initialTextRegExp = new RegExp('select serviceaccount', 'i');
+
+const middleware = [thunk];
+const mockStore = configureStore(middleware);
+
+jest
```

Comment:
`spyOn` and other setup should be called from inside an `it` to allow for proper cleanup between tests

if there's common code you want to execute in all of the tests you can use a `beforeEach`